packaging/test_plugin_storage_tiering.py

```diff
@@ -806,4 +832,90 @@ def test_directory_registration(self):
+            admin_session.assert_icommand('iadmin rmresc ufs1')
+            admin_session.assert_icommand('iadmin rum')
+
+    def test_put_gt_max_row_cnt(self):
+        with storage_tiering_configured():
+            with session.make_session_for_existing_admin() as admin_session:
+                file_count = 257
+                dirname = 'test_put_gt_max_row_cnt'
+                lib.make_large_local_tmp_dir(dirname, file_count, 1)
+                admin_session.assert_icommand(['iput', '-R', 'ufs0', '-r', dirname], 'STDOUT_SINGLELINE', ustrings.recurse_ok_string())
+
+                # stage to tier 1, everything should have been tiered out
+                sleep(5)
+                admin_session.assert_icommand('irule -r irods_rule_engine_plugin-storage_tiering-instance -F /var/lib/irods/example_tiering_invocation.r')
+                admin_session.assert_icommand('iqstat', 'STDOUT_SINGLELINE', 'apply')
+                sleep(80)
+                admin_session.assert_icommand('iqstat', 'STDOUT_SINGLELINE', 'No')
+                admin_session.assert_icommand(['ils', '-l', dirname], 'STDOUT_SINGLELINE', 'ufs1')
+                admin_session.assert_icommand_fail(['ils', '-l', dirname], 'STDOUT_SINGLELINE', 'ufs0')
+
+                # cleanup
+                admin_session.assert_icommand(['irm', '-f', '-r', dirname])
+                shutil.rmtree(dirname, ignore_errors=True)
+
+    def test_put_max_row_cnt(self):
+        with storage_tiering_configured():
+            with session.make_session_for_existing_admin() as admin_session:
+                file_count = 256
+                dirname = 'test_put_max_row_cnt'
+                lib.make_large_local_tmp_dir(dirname, file_count, 1)
+                admin_session.assert_icommand(['iput', '-R', 'ufs0', '-r', dirname], 'STDOUT_SINGLELINE', ustrings.recurse_ok_string())
+
+                # stage to tier 1, everything should have been tiered out
+                sleep(5)
+                admin_session.assert_icommand('irule -r irods_rule_engine_plugin-storage_tiering-instance -F /var/lib/irods/example_tiering_invocation.r')
+                admin_session.assert_icommand('iqstat', 'STDOUT_SINGLELINE', 'apply')
+                sleep(80)
+                admin_session.assert_icommand('iqstat', 'STDOUT_SINGLELINE', 'No')
+                admin_session.assert_icommand(['ils', '-l', dirname], 'STDOUT_SINGLELINE', 'ufs1')
+                admin_session.assert_icommand_fail(['ils', '-l', dirname], 'STDOUT_SINGLELINE', 'ufs0')
+
+                # cleanup
+                admin_session.assert_icommand(['irm', '-f', '-r', dirname])
+                shutil.rmtree(dirname, ignore_errors=True)
+
+    def test_put_object_limit_lt(self):
+        with storage_tiering_configured():
+            with session.make_session_for_existing_admin() as admin_session:
+                admin_session.assert_icommand('imeta add -R ufs0 irods::storage_tiering::object_limit 257')
+
+                file_count = 258
```

Comment:
these are strangely just above 256... can we get a comment as to why?

Reply:
Sure thing. For completeness, I'll also say here that I made it this way so that we are forced to use continueInx (which happens with result sets >256 objects).